pkg/deploy/api/types.go

```diff
@@ -183,6 +183,12 @@ const (
 	// new deployment.
 	// TODO: This should be made public upstream.
 	DesiredReplicasAnnotation = "kubectl.kubernetes.io/desired-replicas"
+	// DeploymentStatusReasonAnnotation represents the reason for deployment being in a given state
+	// Used for specifying the reason for cancellation or failure of a deployment
+	DeploymentStatusReasonAnnotation = "openshift.io/deployment.status-reason"
+	// DeploymentCancelledAnnotation indicates that the deployment has been cancelled
+	// The annotation value does not matter and its mere presence indicates cancellation
+	DeploymentCancelledAnnotation = "openshift.io/deployment.cancelled"
```

Comment:
As of now, the reason will either be "Cancelled by user" or "Cancelled because it exceeded maximum allowed time". The text is not the actual text but the gist of it.